lib/commons/builder/wikidata_queries.rb

```diff
@@ -31,213 +38,39 @@ def query_legislative(position_item_id:, house_item_id:, term_item_id: nil, star
-        FILTER NOT EXISTS { ?legislature wdt:P576 ?legislatureEnd . FILTER (?legislatureEnd < NOW()) }
-        #{label_service}
-      } ORDER BY ?primarySort ?country ?adminAreaType ?legislature ?legislaturePost
-    SPARQL
+    query = templated_query('legislative_index')
+    query.result binding
   end
 
   def query_legislative_index_terms(*houses)
-    houses = houses.map { |house| "wd:#{house}" }.join(' ')
-    <<~SPARQL
-      SELECT DISTINCT ?house ?houseLabel ?legislature ?legislatureLabel ?term ?termLabel ?termStart ?termEnd WHERE {
-        VALUES ?house { #{houses} }
-        ?house (p:P361/ps:P361)* ?legislature .
-            ?baseTerm p:P31|p:P279 [ ps:P279|ps:P31 wd:Q15238777 ; pq:P642 ?legislature ] .
-            OPTIONAL { ?subTerm wdt:P31 ?baseTerm }
-
-        BIND(COALESCE(?subTerm, ?baseTerm) AS ?term)
-
-        OPTIONAL { ?term (wdt:P580|wdt:P571) ?termStart. }
-        OPTIONAL { ?term (wdt:P582|wdt:P576) ?termEnd. }
-        OPTIONAL { ?term (wdt:P155|wdt:P1365) ?termReplaces }
-        OPTIONAL { ?term (wdt:P156|wdt:P1366) ?termReplacedBy }
-
-        FILTER (!BOUND(?termEnd) || ?termEnd > NOW())
-        FILTER (!BOUND(?termReplacedBy))
-        #{label_service}
-      } ORDER BY ?termStart ?term
-    SPARQL
+    query = templated_query('legislative_index_terms')
+    query.result binding
   end
 
   def query_executive_index(country)
-    country = "wd:#{country}" unless country.start_with?('wd:')
-    <<~SPARQL
-      SELECT DISTINCT ?executive ?executiveLabel ?adminArea ?adminAreaLabel ?adminAreaType ?adminAreaTypeLabel ?position ?positionLabel {
-        {
-          #{select_admin_areas_for_country(country)}
-        }
-
-        OPTIONAL {
-          {
-            ?position p:P1001 [ wikibase:rank ?appliesToJurisdictionRank ; ps:P1001 ?adminArea ] ;
-              wdt:P31/wdt:P279* wd:Q4164871 .
-            FILTER (?appliesToJurisdictionRank != wikibase:DeprecatedRank)
-            FILTER EXISTS {
-              VALUES ?positionSuperclass { wd:Q2285706 wd:Q30461 }
-              ?position wdt:P279* ?positionSuperclass .
-            }
-          } UNION {
-            ?adminArea wdt:P1313 ?position
-          }
+    query = templated_query('executive_index')
+    query.result binding
+  end
 
-          OPTIONAL {
-            ?position wdt:P361 ?executive .
-            # Exclude executives that are privy councils
-            FILTER NOT EXISTS { ?executive wdt:P31/wdt:P279* wd:Q6528244 }
-            # Exclude executives which aren't direct parents of the position
-            FILTER NOT EXISTS { ?position wdt:P361 ?other . ?other wdt:P361+ ?executive }
-          }
-        }
+  private
 
-        #{label_service}
-      } ORDER BY ?primarySort ?country ?adminAreaType ?executive ?position
-    SPARQL
+  def templated_query(name)
+    @templated_queries ||= {}
+    @templated_queries[name] ||= ERB.new File.open(File.join(File.dirname(__FILE__), 'queries', name + '.rq.erb')).read
```

Comment:
I feel this could be tidier and/or more idiomatic, but I'm not sure how.

Reply:
Using `Pathname` and `__dir__` would help a bit in this case, because you could do:

```
Pathname.new(__dir__).join('queries', name + '.rq').read
```

Reply:
Awesome, thank you :-)

Reply:
Thinking about how you might test this easily, I think I'd also separate out the functionality of "rendering a template that's a string" from "render a template that I find from the filesystem". (It would be good for the tests of template rendering to have small example queries, rather than the huge ones that we have in the repository anyway, without having to worry about creating a file for them, etc.)

Reply:
This has been done (`templated_query` now loads the template as a string and then calls `templated_query_from_string`)